rust/codegen/src/main.rs

```diff
@@ -0,0 +1,22 @@
+extern crate protoc_rust;
+
+use std::env;
+use std::vec::Vec;
+use protoc_rust::Customize;
+
+fn main() {
+	let args: Vec<_> = env::args().collect();
+	let mut root = "../..";
+	if args.len() > 1 {
+		root = &args[1];
+	}
+
+	protoc_rust::run(protoc_rust::Args {
+	    out_dir: &format!("{}{}", root, "/rust/src"),
+	    input: &[&format!("{}{}", root, "/proofs.proto")],
```

Comment:
For windows compatibility, you might want to use `env::var("OUT_DIR")` (see [the codegen section of the book for more on that](https://doc.rust-lang.org/cargo/reference/build-scripts.html#case-study-code-generation)) to get the root and [PathBuf](https://doc.rust-lang.org/std/path/struct.PathBuf.html) to handle path separators properly, and then the `to_str` method to format it as an optional `&str` (it's an `Option` because windows paths are utf-16 and `str` is guaranteed to be valid utf-8, and maybe something got messed up.)

Reply:
I kind of explicitly don't support windows for dev environments (you can run the code, but not necessarily contribute). Good tip here, but I'll leave out windows